src/test/java/org/assertj/core/error/ShouldHaveSameFileSystem_create_Test.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.error;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.error.ShouldHaveSameFileSystem.shouldHaveSameFileSystem;
+import static org.assertj.core.error.ShouldHaveSameFileSystem.shouldHaveSameFileSystemAsPath;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.nio.file.FileSystem;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.assertj.core.description.TextDescription;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for {@link ShouldHaveSameFileSystem}.
+ *
+ * @author Ashley Scopes
+ */
+class ShouldHaveSameFileSystem_create_Test {
```

Comment:
split tests in `ShouldHaveFileSystem_create_Test` and `ShouldHaveSameFileSystem_create_Test`

Reply:
Fixed and split up